### ISSUE TITLE
Installation errors out on NPM task since 4.7.0

### ISSUE THREAD

**Issue by joeriharleman** [DataDog/ansible-datadog#331]
Dear Datadog Ansible role maintainers,

Since an hour ago or so our deployment pipeline is failing on our test environment due to what seems to be the release of version 4.7.0 of the role. I haven't been able to fully debug the situation and I don't know what information would be required for you guys to provide assistance (or fix the issue if it is a bug). The thing that confuses me is that we have 2 Windows Server 2019 installations, one of which successfully installs Datadog using the new version of the role, the other receives an error.

Here is the error we're receiving:

```
PLAY [Install Datadog] ********************************************************* 
TASK [Gathering Facts] ********************************************************* 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Gather Ansible Facts] ********************************** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Check if OS is supported] ****************************** 
included: /opt/home/some_user/.ansible/roles/datadog.datadog/tasks/os-check.yml for masked-hostname.domain 
TASK [datadog.datadog : Fail if OS is not supported] *************************** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : Set Facts for Datadog Agent Major Version] ************* 
included: /opt/home/some_user/.ansible/roles/datadog.datadog/tasks/set-parse-version.yml for masked-hostname.domain 
TASK [datadog.datadog : Convert datadog_agent_major_version to string] ********* 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : include_tasks] ***************************************** 
included: /opt/home/some_user/.ansible/roles/datadog.datadog/tasks/parse-version.yml for masked-hostname.domain 
TASK [datadog.datadog : Parse Agent version] *********************************** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Set version vars] ************************************** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Fill empty version epoch with default] ***************** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Fill empty version release with default] *************** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Stop play if datadog_agent_version and datadog_agent_major_version are not compatible] *** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : Set datadog_agent_major_version to deduced value from datadog_agent_version] *** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Set OS-specific versions] ****************************** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Set Agent default major version] *********************** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : Debian Install Tasks] ********************************** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : RedHat Install Tasks] ********************************** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : Suse Install Tasks] ************************************ 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : Windows Install Tasks] ********************************* 
included: /opt/home/some_user/.ansible/roles/datadog.datadog/tasks/pkg-windows.yml for masked-hostname.domain 
TASK [datadog.datadog : Fail if Agent 5] *************************************** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : Check currently installed Agent version] *************** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Initialize skip install flag to false] ***************** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Set skip install flag if version already installed] **** 
ok: [masked-hostname.domain] 
TASK [datadog.datadog : Download windows datadog agent 614 fix script] ********* 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : Run 6.14.0/1 PowerShell fix] *************************** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : include_tasks] ***************************************** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : include_tasks] ***************************************** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : show URL var] ****************************************** 
skipping: [masked-hostname.domain] 
TASK [datadog.datadog : Set windows NPM installed] ***************************** 
fatal: [masked-hostname.domain]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'network_config' is undefined\n\nThe error appears to be in '/opt/home/some_user/.ansible/roles/datadog.datadog/tasks/pkg-windows.yml': line 59, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n## must be prior to `pkg-windows-opts.yml`, because the variable is used inside\n- name: Set windows NPM installed\n  ^ here\n"} 
```

Role configuration:
```yaml
---
- name: Install Datadog
  hosts: "masked-hostname.domain"
  roles:
  - { role: datadog.datadog }
  vars:
    datadog_site: "datadoghq.eu"
    datadog_config:
      expvar_port: "4000"
      cmd_port: 4001
      gui_port: 4002
      clc_runner_port: 4005
      cluster_checks:
        clc_runners_port: 4005
      dogstatsd_stats_port: 4000
      hostname_fqdn: true
      tags:
      - "env:{{ environment_name }}"
    datadog_checks:
      http_check:
        init_config:
        instances:
        - name: Web Viewer
          url: <masked url>
        - name: Token Service
          url: <masked url>
        - name: Status Page
          url: <masked url>
      tcp_check:
        init_config:
        instances:
        # Local ports
        - name: Client
          host: localhost
          port: <masked port>
        - name: Service
          host: localhost
          port: <masked port>
      windows_service:
        init_config:
        instances:
        - services:
          - some masked services

```

**Comment by bkabrda** [comment]
Hi 👋 thanks for the report, I'll look into this and update this issue once I found out what might be going wrong.

**Comment by lognarly** [comment]
This is happening on Linux as well. I opened issue 330 for this at almost the exact same time.

**Comment by joeriharleman** [comment]
> This is happening on Linux as well. I opened issue 330 for this at almost the exact same time.

Oh, we must've made the issue around the same time then, hah. I made sure to check before creating this issue. Either way, updated the title to reflect that information.

**Comment by joeriharleman** [comment]
Looks to be caused by PR #326

**Comment by bkabrda** [comment]
Hi everyone, thanks for your patience. We've identified several different issues in #326. We'll be reverting and releasing a bugfix release 4.7.1 soon. Then we'll properly fix everything and do a new release with fixed Windows NPM support.

**Comment by bkabrda** [comment]
4.7.1 is now released, reverting #326: https://github.com/DataDog/ansible-datadog/releases/tag/4.7.1

I'll update/close this issue when we implement and release the fixes we talked about.

**Comment by joeriharleman** [comment]
Awesome, thanks for the excellent response time!

**Comment by bkabrda** [comment]
Hey 👋 we've just released version 4.8.0 which re-adds windows NPM and thus should fix your problems. See https://github.com/DataDog/ansible-datadog#system-probe for updated information on sysprobe/network_config setup.

I'm closing this issue. If you have any problems, feel free to open a new one. Thanks!
